internal/setup/setup.go

```diff
@@ -142,3 +143,36 @@ func GetConditionStatus(j *batchv1.Job) (corev1.ConditionStatus, string) {
 
 	return corev1.ConditionFalse, ""
 }
+
+// AppendAssignmentEnv appends an environment variable for each trial assignment
+func AppendAssignmentEnv(t *redskyv1beta1.Trial, env []corev1.EnvVar) []corev1.EnvVar {
+	for _, a := range t.Spec.Assignments {
+		name := strings.ReplaceAll(strings.ToUpper(a.Name), ".", "_")
+		env = append(env, corev1.EnvVar{Name: name, Value: a.Value.String()})
+	}
+
+	return env
+}
+
+// AppendPrometheusEnv appends environment variables to help reference the built in Prometheus
+func AppendPrometheusEnv(t *redskyv1beta1.Trial, env []corev1.EnvVar) []corev1.EnvVar {
+	for i := range t.Spec.SetupTasks {
+		if IsPrometheusSetupTask(&t.Spec.SetupTasks[i]) {
+			url := "http://prometheus:9091/metrics/trial/" + t.Name
```

Comment:
The url here needs to change to be in line with what we deploy -- `redsky-${namespace}-prometheus`